server/service/device_client.go

```diff
@@ -76,7 +76,7 @@ func (dc *DeviceClient) request(verb, pathFmt, token, query string, params inter
 			return reqErr
 		}
 
-		delay := time.Duration(attempt) * time.Second
+		delay := time.Duration(attempt*attempt) * time.Second
```

Comment:
Yeah as Roberto mentions, this is not a backoff due to a server experiencing issues, this is a retry around the time the device token gets invalidated, to address this issue: https://github.com/fleetdm/fleet/issues/13635. You can see on line 70 above that it will retry only if the error is `ErrUnauthenticated`, not any 500 server issue.

For that reason, it shouldn't be modified to an exponential backoff - we want it to succeed as quickly as possible.

Reply:
@mna My point is that the retries should back off if they keep failing. The first retry will still happen in 1 second (1*1=1). But if that fails (meaning 2 requests already failed), doesn't that signal a potential server issue, and we should start backing off?

Reply:
> doesn't that signal a potential server issue, and we should start backing off?

@getvictor No, in that case if it keeps failing (and only if it keeps failing with `ErrUnauthenticated`, if there's any other sort of server error we don't retry), it means the server hasn't yet rotated the token but we know it is just about to do it. We want to keep the retries both short in total time and catch the new token as quickly as possible.